### PR TITLE
Remove legacy local, remote and virtual repos

### PR DESCRIPTION
**Pull request by danielmkn**
#375 
Remove legacy repositories code - local, remote and virtual
Fix the documentation accordingly
Fix errors in the doc HCL examples
Remove tests dependency on ARTIFACTORY_USERNAME in some cases
Add some fields in repositories.go (PropagateQueryParams)
Added `AddRetryCondition` for several calls to prevent errors
Removed deprecated legacy tests
Added all the resource examples to sample.tf

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,8 +84,6 @@ ARTIFACTORY_USERNAME=admin
 ARTIFACTORY_ACCESS_TOKEN=<your_access_token>
 TF_ACC=true
 ```
-`ARTIFACTORY_USERNAME` is not used in authentication, but used in several tests, related to replication functionality.
```
Comment (chb0github) [inline README.md:87]: This is an unrelated change. Revert it please

File: docs/resources/artifactory_local_repository.md [hunk 2]
```diff
@@ -6,9 +6,8 @@ Provides an Artifactory local repository resource. This can be used to create an
 
 ```hcl
 # Create a new Artifactory local repository called my-local
-resource "artifactory_local_repository" "my-local" {
+resource "artifactory_local_npm_repository" "my-local" {
```
Comment (chb0github) [inline docs/resources/artifactory_local_repository.md:9]: Should this file even exist anymore? The resource is gone

File: docs/resources/artifactory_remote_cargo_repository.md [hunk 3]
```diff
@@ -3,6 +3,17 @@
 Provides an Artifactory remote `cargo` repository resource. This provides cargo specific fields and is the only way to get them
 Official documentation can be found [here](https://www.jfrog.com/confluence/display/JFROG/Cargo+Registry)
 
+### Passwords
```
Comment (chb0github) [inline docs/resources/artifactory_remote_cargo_repository.md:6]: This seems to be an unrelated change

File: docs/resources/artifactory_remote_cargo_repository.md [hunk 4]
```diff
@@ -13,6 +24,7 @@ for brevity sake, only cargo specific fields are included; for other fields see
 resource "artifactory_remote_cargo_repository" "my-remote-cargo" {
   key                 = "my-remote-cargo"
   anonymous_access    = true
+  url                 = "https://github.com/"
```
Comment (chb0github) [inline docs/resources/artifactory_remote_cargo_repository.md:27]: I am not against these sorts of changes as part of this PR, but an explanation is in order

File: docs/resources/artifactory_remote_cargo_repository.md [hunk 5]
```diff
@@ -31,7 +43,7 @@ Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/c
 * `project_environments` - (Optional) Project environment for assigning this repository to. Allow values: "DEV" or "PROD"
 * `url` - (Required) the remote repo URL. You kinda don't have a remote repo without it
 * `username` - (Optional)
-* `password` - (Optional)
+* `password` - (Optional) Requires password encryption to be turned off `POST /api/system/decrypt`
```
Comment (chb0github) [inline docs/resources/artifactory_remote_cargo_repository.md:46]: What's with this last bit?

File: docs/resources/artifactory_remote_docker_repository.md [hunk 6]
```diff
@@ -3,6 +3,17 @@
 Provides an Artifactory remote `docker` repository resource. This provides docker specific fields and is the only way to
 get them
 
+### Passwords
+Passwords can only be used when encryption is turned off (https://www.jfrog.com/confluence/display/RTF/Artifactory+Key+Encryption).
+Since only the artifactory server can decrypt them it is impossible for terraform to diff changes correctly.
+
+To get full management, passwords can be decrypted globally using `POST /api/system/decrypt`. If this is not possible,
```
Comment (chb0github) [inline docs/resources/artifactory_remote_docker_repository.md:10]: Is it possible to use this to mitigate our password issues @alexhung 

Also, why is this documentation change here? What does it have to do with docker and repos?

**Review by chb0github — CHANGES_REQUESTED** [review]
Ok, I don't understand what's up with all this password stuff and related documentation changes. Besides that, it's fine

**Review by alexhung — CHANGES_REQUESTED** [review]
Some minor changes needed for repos related code.

Password related changes should be part of separate PR.

File: docs/resources/artifactory_remote_cargo_repository.md (hunk 3 above)
Comment (danielmkn) [reply docs/resources/artifactory_remote_cargo_repository.md]: removed from this PR

File: docs/resources/artifactory_remote_cargo_repository.md (hunk 4 above)
Comment (danielmkn) [reply docs/resources/artifactory_remote_cargo_repository.md:16]: it's a missing required field in the HCL example

File: docs/resources/artifactory_remote_cargo_repository.md (hunk 5 above)
Comment (danielmkn) [reply docs/resources/artifactory_remote_cargo_repository.md]: removed from this PR

File: docs/resources/artifactory_remote_docker_repository.md (hunk 6 above)
Comment (danielmkn) [reply docs/resources/artifactory_remote_docker_repository.md]: removed from this PR

File: pkg/artifactory/repositories.go [hunk 7]
```diff
@@ -525,9 +525,9 @@ var baseRemoteRepoSchema = map[string]*schema.Schema{
 		StateFunc: getMD5Hash,
 	},
 	"proxy": {
-		Type:     schema.TypeString,
-		Optional: true,
-		Computed: true,
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Proxy key from Artifactory Proxies setting",
```
[the remark this answers is not in the document]
Comment (chb0github) [reply pkg/artifactory/repositories.go:530]: agree. Let's keep it but stay disciplined for the future

File: pkg/artifactory/resource_artifactory_permission_target_test.go [hunk 8]
```diff
@@ -9,9 +9,8 @@ import (
 )
 
 const permissionNoIncludes = `
-	//resource "artifactory_local_repository" "{{ .repo_name }}" {
+	//resource "artifactory_local_docker_repository" "{{ .repo_name }}" {
```
[the remark this answers is not in the document]
Comment (chb0github) [reply pkg/artifactory/resource_artifactory_permission_target_test.go:12]: agree

File: README.md (hunk 1 above)
Comment (danielmkn) [reply README.md:87]: removed form the PR

File: docs/resources/artifactory_local_repository.md (hunk 2 above)
Comment (danielmkn) [reply docs/resources/artifactory_local_repository.md:9]: I think we need a separate PR for this?

File: pkg/artifactory/resource_artifactory_permission_target_test.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (danielmkn) [reply pkg/artifactory/resource_artifactory_permission_target_test.go:12]: I'll remove them in the next PR, when doing a cleanup

File: pkg/artifactory/resource_artifactory_permission_target_test.go [hunk 9]
```diff
@@ -24,15 +23,14 @@ const permissionNoIncludes = `
 				}
 			}
 		}
-     //depends_on = [artifactory_local_repository.{{ .repo_name }}]
+     //depends_on = [artifactory_local_docker_repository.{{ .repo_name }}]
```
[the remark this answers is not in the document]
Comment (danielmkn) [reply pkg/artifactory/resource_artifactory_permission_target_test.go:26]: I'll remove them in the next PR, when doing a cleanup

File: sample.tf [hunk 10]
```diff
@@ -12,50 +12,152 @@ provider "artifactory" {
...
-  max_unique_snapshots       = 10
-  force_nuget_authentication = true
+resource "artifactory_local_bower_repository" "bower-local" {
+  key         = "bower-local"
+  description = "Repo created by Terraform Provider Artifactory"
 }
 
-resource "artifactory_local_generic_repository" "my-generic-local" {
-  key                 = "my-generic-local"
+#resource "artifactory_local_cargo_repository" "cargo-local" {
```
[the remark this answers is not in the document]
Comment (danielmkn) [reply sample.tf:20]: I'll remove them in the next PR, when doing a cleanup

File: sample.tf [hunk 11]
```diff
@@ -12,50 +12,152 @@ provider "artifactory" {
...
+  key         = "composer-local"
+  description = "Repo created by Terraform Provider Artifactory"
+}
+
+resource "artifactory_local_conan_repository" "conan-local" {
+  key         = "conan-local"
+  description = "Repo created by Terraform Provider Artifactory"
+}
+
+#resource "artifactory_local_conda_repository" "conda-local" {
```
[the remark this answers is not in the document]
Comment (danielmkn) [reply sample.tf:45]: oh, this one wasn't implemented when I was working on the PR. I need to uncomment it

**Review by alexhung — CHANGES_REQUESTED** [review]
Needs to update `go.mod` and `main.go` with `v5`. CHANGELOG.md needs update too.

**Review by maheshjfrog — DISMISSED** [review]

**Review by dasmanas — APPROVED** [review]

**Review by dasmanas — DISMISSED** [review]

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,8 +1,16 @@
+<<<<<<< HEAD
```
Comment (chb0github) [inline CHANGELOG.md:1]: YOu have a merge line here. Please remove it

File: pkg/artifactory/resource_artifactory_permission_target_test.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (chb0github) [reply pkg/artifactory/resource_artifactory_permission_target_test.go:26]: Please create yourself a ticket now

File: pkg/artifactory/resource_artifactory_permission_target_test.go [hunk 13]
```diff
@@ -196,7 +192,7 @@ func TestAccPermissionTarget_full(test *testing.T) {
 
 func TestAccPermissionTarget_addBuild(t *testing.T) {
 	_, permFqrn, permName := mkNames("test-perm", "artifactory_permission_target")
-	//_, _, repoName := mkNames("test-perm-repo", "artifactory_local_repository")
+	//_, _, repoName := mkNames("test-perm-repo", "artifactory_local_generic_repository")
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_permission_target_test.go:195]: I can't imagine why not delete these now. We've been asked to hold off on this a bit. So, why not clean up?

**Review by chb0github — CHANGES_REQUESTED** [review]
You have a merge line here you can remove

**Review by alexhung — CHANGES_REQUESTED** [review]
Same feedback as @chb0github

**Comment by chb0github** [comment]
https://github.com/jfrog/terraform-provider-artifactory/discussions/384

File: CHANGELOG.md (hunk 12 above)
Comment (danielmkn) [reply CHANGELOG.md:1]: OMG, sure, sorry :(

File: pkg/artifactory/resource_artifactory_permission_target_test.go (hunk 13 above)
Comment (danielmkn) [reply pkg/artifactory/resource_artifactory_permission_target_test.go]: removed

**Review by alexhung — DISMISSED** [review]

**Review by chb0github — APPROVED** [review]
